client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -15,84 +16,78 @@ import { recordTracksEvent } from 'state/analytics/actions';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSite, isCurrentSitePlan } from 'state/sites/selectors';
 import { plansList, PLAN_FREE, PLAN_PERSONAL } from 'lib/plans/constants';
-import storeTransactions from 'lib/store-transactions';
-import upgradesActions from 'lib/upgrades/actions';
+import { storedCardPayment } from 'lib/store-transactions';
 import { getStoredCards } from 'state/stored-cards/selectors';
 import QueryStoredCards from 'components/data/query-stored-cards';
-import TransactionStepsMixin from 'my-sites/upgrades/checkout/transaction-steps-mixin';
-import { fillInAllCartItemAttributes } from 'lib/cart-values';
-import { add, remove, getItemForPlan } from 'lib/cart-values/cart-items';
+import { emptyCart } from 'lib/cart-values';
+import { add } from 'lib/cart-values/cart-items';
 import { submitTransaction } from 'lib/upgrades/actions/checkout';
 
-//use the redux store instead:
-import productsListFactory from 'lib/products-list';
-const productsList = productsListFactory();
+const debug = debugFactory( 'calypso:domain-to-plan-nudge' );
 
-const DomainToPlanNudge = React.createClass( {
+class DomainToPlanNudge extends Component {
 
-	//verify if we need this mixin
-	mixins: [ TransactionStepsMixin ],
+	constructor() {
+		super( ...arguments );
```

Comment:
Why not do the typical:

```js
constructor( props ) {
   super( props );
}
```
? And btw, if we don't need access to `this.props` in the constructor, we don't have to pass the `props` down to the `super` call.